test/unit/collection-view.spec.js

```diff
@@ -1276,4 +1276,312 @@ describe('collection view', function() {
       expect(this.childView.$el).to.contain.$text('bar');
     });
   });
+
+  describe("filtering", function () {
+    beforeEach(function() {
+      var spec = this;
+
+      this.passModel = new Backbone.Model({foo: true});
+      this.failModel = new Backbone.Model({foo: false});
+      this.collection = new Backbone.Collection();
+
+      this.EmptyView = Backbone.Marionette.ItemView.extend({
+        template: function () { return 'empty'; }
+      });
+
+      this.filter = this.sinon.spy(function(child) {
+        return child.get('foo');
+      });
+
+      this.inverseFilter = this.sinon.spy(function (child) {
+        return !spec.filter(child);
+      });
+
+      this.CollectionView = this.MockCollectionView.extend({
+        emptyView: this.EmptyView,
+        filter: this.filter,
+        collection: this.collection
+      });
+    });
+
+    describe("testChild", function() {
+      it("returns the result of the filter", function() {
+        var collectionView = new this.CollectionView();
+        expect(collectionView.testChild(this.passModel)).to.be.true;
+        expect(collectionView.testChild(this.failModel)).to.be.false;
+      });
+
+      it("will prefer to use the filter supplied at construction", function() {
+        var collectionView = new this.CollectionView({
+          filter: this.inverseFilter
+        });
+        expect(collectionView.testChild(this.passModel)).to.be.false;
+        expect(collectionView.testChild(this.failModel)).to.be.true;
+      });
+
+      it("always returns true when no filter is supplied", function (){
+        var collectionView = new this.MockCollectionView();
+        expect(collectionView.testChild(this.passModel)).to.be.true;
+        expect(collectionView.testChild(this.failModel)).to.be.true;
+      });
+    });
+
+    describe("setFilter", function() {
+      beforeEach(function () {
+        this.collection.add(this.passModel);
+        this.collection.add(this.failModel);
+        this.collectionView = new this.CollectionView();
+        this.sinon.spy(this.collectionView, 'render');
+      });
+
+      describe("when the view has not been rendered", function () {
+        beforeEach(function() {
+          this.collectionView.setFilter(this.inverseFilter);
+        });
+
+        it("should set the filter", function() {
+          expect(this.collectionView.filter).to.equal(this.inverseFilter);
+        });
+
+        it("should not render the view", function() {
+          this.collectionView.setFilter(this.inverseFilter);
+          expect(this.collectionView.render).not.to.have.been.called;
```

Comment:
we should put a 

.called.once
.called.with

on all of these. These assertions have caught some nasty issues in the past for us. 

remember @thejameskyle  :dancer: